### ISSUE TITLE
[Feat]: Multiple selection combobox 

### ISSUE THREAD

**Issue by kzarora** [adobe/spectrum-web-components#4777]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Picker components

### Description of the requested feature

The user should be able to select more than one option in the combobox drop-down as shown in the image below.

### Mockups or screenshots

<img width="229" alt="Screenshot 2024-09-23 at 2 52 57 PM1 copy" src="https://github.com/user-attachments/assets/4de9a51c-7565-404c-b521-3aa9b688bdb1">

**Comment by najikahalsema** [comment]
@kzarora hi there, question for you--what is the experience you want to achieve here? If you're entering specific text into an input field,I am unsure why multiple selections could be possible. In other words, in the example you provided, if you have inputted "Adobe InDesign", why would you be able to select menu items that don't match that string? 

perhaps something [like this experience](https://glideapps.github.io/glide-data-grid/?path=/story/extra-packages-cells--custom-cell-editing) is what you'd be looking for instead? 

![image](https://github.com/user-attachments/assets/e9054bd4-0782-4b06-a6a2-6e32027cf11f)
![image](https://github.com/user-attachments/assets/8f5b5c14-be18-4312-bcb2-8239e5dd7c48)

**Comment by kzarora** [comment]
Yes I am more concerned about the multiple selection options instead of typing in search for the element.
Much like 
![image](https://github.com/user-attachments/assets/f1bda432-4f41-4426-9f7d-823018698434)

**Comment by najikahalsema** [comment]
@kzarora in that case, could you use the picker instead?

**Comment by lschierer** [comment]
can you do this with picker?  I do not see an obvious way in the documentation...

**Comment by najikahalsema** [comment]
@lschierer I'm sorry, you're right. I'll explain.

Currently, we don't have something like the experience i linked in my initial comment; however, Spectrum is currently working on a new component that would allow for multiple tags to be inputted into a textfield as it works in the example I showed.

I erroneously believed that the Picker had multi-select functionality--this is not true. _Menu_ does, but that still doesn't give you the ability to select multiple items and see them displayed in the `<input>` field, which is what is happening here:
![image](https://github.com/user-attachments/assets/bcb437cb-cba9-4460-b45e-6a859a9d9e4b)

I think this use case is interesting and I'll mention it to the design and development teams to discuss options. Thank you for your patience!
